### PR TITLE
Exclude WALinuxAgent from vm update scripts

### PR DESCRIPTION
**Pull request by s-amann**
### Which issue this PR addresses:

[WI](https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/16113483)

Fixes hung up WALinuxAgent due to being updated

### What this PR does / why we need it:

Updating the WALinuxAgent will stop the agent causing the script to not complete.

### Test plan for issue:

N/A

### Is there any documentation that needs to be updated for this PR?

N/A

### CODE REVIEW

File: pkg/deploy/generator/scripts/rpVMSS.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-yum -y update
+yum -y -x WALinuxAgent update
```
Comment (cadenmarchese) [inline pkg/deploy/generator/scripts/rpVMSS.sh:1]: There's a yum update later in the script as well (after repositories are added), should we exclude there as well?

**Review by cadenmarchese — COMMENTED** [review]
Looking good, one question:

File: pkg/deploy/generator/scripts/rpVMSS.sh (hunk 1 above)
Comment (s-amann) [reply pkg/deploy/generator/scripts/rpVMSS.sh:1]: I only see two yum installs in the remainder of the files

Comment (jharrington22) [reply pkg/deploy/generator/scripts/rpVMSS.sh:1]: I don't see the other update there is only an install of a specific rpm [here](https://github.com/Azure/ARO-RP/pull/2512/files#diff-a7a118c3717408a3425f9209b28c891edad1c68d9b9a67cb41b18545f8ce37e3R16) and then a list of specific packages on [here](https://github.com/Azure/ARO-RP/pull/2512/files#diff-a7a118c3717408a3425f9209b28c891edad1c68d9b9a67cb41b18545f8ce37e3R74) am I missing it?

Comment (cadenmarchese) [reply pkg/deploy/generator/scripts/rpVMSS.sh:1]: You're right, they are not updates. LGTM!

**Review by jharrington22 — APPROVED** [review]
/lgtm

**Comment by s-amann** [comment]
> 

I believe the idea behind us excluding the WALinuxAgent from our update is Azure should be taking care of which version is to be used by a new VMSS. I have reached out to MSFT to get clarification on this.

**Comment by jhoreman** [comment]
We should consider addressing the other potential race condition of the reboot command as well. Proposing we increase the sleep count on the last line to 120 (up from 30)

Line #625: (sleep 120; reboot) &

**Comment by s-amann** [comment]
> We should consider addressing the other potential race condition of the reboot command as well. Proposing we increase the sleep count on the last line to 120 (up from 30)
> 
> Line #625: (sleep 120; reboot) &

done

**Review by jhoreman — APPROVED** [review]
LGTM thanks

**Review by jhoreman — APPROVED** [review]
lgtm

**Review by cadenmarchese — APPROVED** [review]

**Pull request merged**